js/web/lib/onnxjs/backends/webgl/ops/reshape-packed.ts

```diff
@@ -32,18 +31,22 @@ export class WebGLReshapePacked extends Reshape implements WebGLOperator {
     // the same between input shape and output shape, the packed reshape can be
     // treated as no-op.
     const originInputShape = inputs[0].dims;
-    const inputShape3D = processDims3D(inputs[0].dims);
+    this.inputShape3D = processDims3D(inputs[0].dims);
     let inputLayout: TextureLayout;
-    if (originInputShape.length === 3) {
-      inputLayout = handler.getOrCreateTextureLayout(inputs[0], 4, true, originInputShape, true);
-    } else {
+    inputLayout = handler.getOrCreateTextureLayout(inputs[0], 4, true, originInputShape, true);
+    if (originInputShape.length !== 3) {
+      const originalInputLayout = inputLayout;
```

Comment:
>const originalInputLayout = inputLayout; [](http://example.com/codeflow?start=6&length=40)

in the future, we may want to redesign this and discouple texture layout with tensor, as tensor is just a logic representation and can have multiple layout associated with it. Then it can save us some hussle of creating those intermediate texturelayout, just to have a different tensor id as a key to the texture cahce. 

Maybe we can add a TODO as a comment here mentioning the possible optimization in the future.

Reply:
Will do it in another PR